doc/TE.md

```diff
@@ -343,6 +343,52 @@ append.  This could be used, for example to generally allow log files to be
 both writable and appendable, while allowing exceptions for certain append-only
 log scenarios.
 
+## Type casting
+
+To help protect against incorrect AV rules there are rules in place that
```

Comment:
I'm conflicted on this paragraph.  On one hand, this is a really key motivator for the feature.  On the other hand, there's not really something all that special about AV rules.  Type casting works for any function call.

I might lead with something about how Cascade does strict type checking on all function arguments in order to ensure correctness or something like that.  Then bring up the specific AV rules example as the most likely use case here.  Then finally maybe cite an example of when you would want to use this (probably /proc is the simplest case)